src/OrleansProviders/Streams/Common/PooledCache/IObjectPool.cs

```diff
@@ -41,6 +47,11 @@ public abstract class PooledResource<T> : IDisposable
         public IObjectPool<T> Pool { set { pool = value; } }
 
         /// <summary>
+        /// Pooled resource size
+        /// </summary>
+        public virtual int SizeInByte { get {return Marshal.SizeOf(this);} }
```

Comment:
shouldn't be calculating these sizes per call at runtime.  As from other post, I think we should avoid size in the general case.